test/integration/exchanged_token_integration_test.cc

```diff
@@ -0,0 +1,418 @@
+// backend).
+
+#include "fmt/printf.h"
+#include "gmock/gmock.h"
+#include "include/istio/utils/attribute_names.h"
+#include "mixer/v1/mixer.pb.h"
+#include "src/envoy/utils/filter_names.h"
+#include "test/integration/http_protocol_integration.h"
+
+using ::google::protobuf::util::error::Code;
+using ::testing::Contains;
+using ::testing::Not;
+
+namespace Envoy {
+namespace {
+
+// An example exchanged token
+constexpr char kExchangedToken[] =
+    "eyJhbGciOiJSUzI1NiIsImtpZCI6IkRIRmJwb0lVcXJZOHQyenBBMnFYZkNtcjVWTzVaRXI0Un"
+    "pIVV8tZW52dlEiLCJ0eXAiOiJKV1QifQ.eyJhdWQiOiJleGFtcGxlLWF1ZGllbmNlIiwiZW1ha"
+    "WwiOiJmb29AZ29vZ2xlLmNvbSIsImV4cCI6NDY5ODM2MTUwOCwiaWF0IjoxNTQ0NzYxNTA4LCJ"
+    "pc3MiOiJodHRwczovL2V4YW1wbGUudG9rZW5fc2VydmljZS5jb20iLCJpc3Rpb19hdHRyaWJ1d"
+    "GVzIjpbeyJzb3VyY2UuaXAiOiIxMjcuMC4wLjEifV0sImtleTEiOlsidmFsMiIsInZhbDMiXSw"
+    "ib3JpZ2luYWxfY2xhaW1zIjp7ImVtYWlsIjoidXNlckBleGFtcGxlLmNvbSIsImlzcyI6Imh0d"
+    "HBzOi8vYWNjb3VudHMuZXhhbXBsZS5jb20iLCJzdWIiOiJleGFtcGxlLXN1YmplY3QifSwic3V"
+    "iIjoiaHR0cHM6Ly9hY2NvdW50cy5leGFtcGxlLmNvbS8xMjM0NTU2Nzg5MCJ9.mLm9Gmcd748a"
+    "nwybiPxGPEuYgJBChqoHkVOvRhQN-H9jMqVKyF-7ynud1CJp5n72VeMB1FzvKAV0ErzSyWQc0i"
+    "ofQywG6whYXP6zL-Oc0igUrLDvzb6PuBDkbWOcZrvHkHM4tIYAkF4j880GqMWEP3gGrykziIEY"
+    "9g4povquCFSdkLjjyol2-Ge_6MFdayYoeWLLOaMP7tHiPTm_ajioQ4jcz5whBWu3DZWx4IuU5U"
+    "IBYlHG_miJZv5zmwwQ60T1_p_sW7zkABJgDhCvu6cHh6g-hZdQvZbATFwMfN8VDzttTjRG8wuL"
+    "lkQ1TTOCx5PDv-_gHfQfRWt8Z94HrIJPuQ";
+
+// An example token without original_claims
+constexpr char kTokenWithoutOriginalClaims[] =
+    "eyJhbGciOiJSUzI1NiIsImtpZCI6IkRIRmJwb0lVcXJZOHQyenBBMnFYZkNtcjVWTzVaRXI0Un"
+    "pIVV8tZW52dlEiLCJ0eXAiOiJKV1QifQ.eyJhdWQiOiJleGFtcGxlLWF1ZGllbmNlIiwiZW1ha"
+    "WwiOiJmb29AZ29vZ2xlLmNvbSIsImV4cCI6NDY5ODcyNzc2NiwiaWF0IjoxNTQ1MTI3NzY2LCJ"
+    "pc3MiOiJodHRwczovL2V4YW1wbGUudG9rZW5fc2VydmljZS5jb20iLCJpc3Rpb19hdHRyaWJ1d"
+    "GVzIjpbeyJzb3VyY2UuaXAiOiIxMjcuMC4wLjEifV0sImtleTEiOlsidmFsMiIsInZhbDMiXSw"
+    "ic3ViIjoiaHR0cHM6Ly9hY2NvdW50cy5leGFtcGxlLmNvbS8xMjM0NTU2Nzg5MCJ9.FVskjGxS"
+    "cTuNFtKGRnQvQgejgcdPbunCAbXlj_ZYMawrHIYnrMt_Ddw5nOojxQu2zfkwoB004196ozNjDR"
+    "ED4jpJA0T6HP7hyTHGbrp6h6Z4dQ_PcmAxdR2_g8GEo-bcJ-CcbATEyBtrDqLtFcgP-ev_ctAo"
+    "BQHGp7qMgdpkQIJ07BTT1n6mghPFFCnA__RYWjPUwMLGZs_bOtWxHYbd-bkDSwg4Kbtf5-9oPI"
+    "nwJc6oMGMVzdjmJYMadg5GEor5XhgYz3TThPzLlEsxa0loD9eJDBGgdwjA1cLuAGgM7_HgRfg7"
+    "8ameSmQgSCsNlFB4k3ODeC-YC62KYdZ5Jdrg2A";
+
+constexpr char kExpectedPrincipal[] =
+    "https://accounts.example.com/example-subject";
+constexpr char kDestinationNamespace[] = "pod";
+constexpr char kDestinationUID[] = "kubernetes://dest.pod";
+constexpr char kSourceUID[] = "kubernetes://src.pod";
+constexpr char kTelemetryBackend[] = "telemetry-backend";
+constexpr char kPolicyBackend[] = "policy-backend";
+const std::string kHeaderForExchangedToken = "ingress-authorization";
+
+// Generates basic test request header.
+Http::TestHeaderMapImpl BaseRequestHeaders() {
```

Comment:
I'm ok with this for now, but it would be nicer if you refactor these to a share library. Alternative (even better), you can have new test cases in the same test suit. They are not much different than the others.

Reply:
Done: added a check for jwt_headers() size and compare the only one in there.